part3/solvers.py

```
@@ -0,0 +1,57 @@
+    """
+    n = len(A)
+    if len(b) != n:
+        raise ValueError("Kích thước ma trận A và vector b không khớp.")
+
+    for i in range(n):
+        if A[i][i] == 0:
+            raise ValueError(f"Phần tử trên đường chéo chính A[{i}][{i}] bằng 0, không thể chia được.")
+
+    # Tiêu chí 3: Kiểm tra chéo trội và cảnh báo
+    if not is_diagonally_dominant(A):
+        warnings.warn("Ma trận không chéo trội. Phương pháp Gauss-Seidel có thể không hội tụ.")
```

Ruff (0.15.9)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)